### PR TITLE
orientdb 2.2.2

### PR DESCRIPTION
**Pull request by austinsmorris**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

**Comment by austinsmorris** [comment]
This PR makes #1589 obsolete.  That PR should be closed (if this passes)...

**Comment by austinsmorris** [comment]
@MikeMcQuaid @UniqMartin, I could use a little help here.  I was able to work past @arxpoetica's issues with the pid and database locations in #1589.  Now `brew test --sandbox --verbose orientdb` is passing for me locally, but this build it still failing.  Any ideas?

File: Formula/orientdb.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Orientdb < Formula
   desc "Graph database"
   homepage "https://orientdb.com"
-  url "https://orientdb.com/download.php?email=unknown@unknown.com&file=orientdb-community-2.1.16.tar.gz&os=mac"
-  version "2.1.16"
-  sha256 "41ad0db53c418459d0efbf6a7f7e2b39f48467f1ec582efa925ceb38de3f3cc6"
+  url "http://orientdb.com/download.php?email=unknown@unknown.com&file=orientdb-community-2.2.2.tar.gz&os=mac"
```
Comment (UniqMartin) [inline Formula/orientdb.rb]: Please don't switch back to insecure HTTP if HTTPS works. It also looks like the URL can be simplified (according to the information in the OrientDB documentation):

```ruby
url "https://orientdb.com/download.php?file=orientdb-community-2.2.2.tar.gz"
```

**Comment by austinsmorris** [comment]
@UniqMartin, url has been updated, but I still have a failing build even though tests pass locally....

**Comment by UniqMartin** [comment]
> url has been updated, but I still have a failing build even though tests pass locally....

Yes, I noticed. I suspect it's an issue on our end, but I haven't had the time yet to investigate. (I hope to be able to do this shortly.) Meanwhile, the URL change caused an audit error. Please try to keep in mind to re-run `brew audit --strict --online <formula>` locally before pushing.

**Comment by austinsmorris** [comment]
@UniqMartin, fixed the audit, thanks.

**Comment by UniqMartin** [comment]
Alright, it's not a problem with our CI. I tried it locally and the formula still has several issues that need to be resolved for this to work.

1. `bin/shutdown.sh` needs to be adjusted (just like `bin/server.sh`). Once I had installed and started the server with `orientdb start`, I was unable to stop it with `orientdb stop`. (I had to kill it manually using Activity Monitor.)

2. The test tries to write to `/usr/local/var` and this is not allowed during the test. All directories where OrientDB needs to write to must be temporarily redirected to subdirectories of `testpath` (the working directory in `test do`). To verify this locally, stop all OrientDB processes and then delete (or move aside) everything that has been created by OrientDB on its first start, particularly `/usr/local/var/db/orientdb/OSystem`. If you then run `brew test --sandbox --verbose orientdb`, the test will fail:

  - `~/Library/Logs/Homebrew/orientdb/sandbox.test.log` will have entries like:

    ```
    Jun 16 16:03:36 sandboxd[136]: java(90130) deny file-write-create /usr/local/var/db/orientdb/OSystem
    Jun 16 16:03:36 sandboxd[136]: java(90130) deny file-write-create /usr/local/var/db/orientdb/OSystem
    ```

 - `/usr/local/var/log/orientdb/orientdb.err` will have entries like:

    ```
    2016-06-16 16:03:36:728 INFO  {db=OSystem} Creating the system database 'OSystem' for current server [OSystemDatabase]
    Exception in thread "main" com.orientechnologies.orient.core.exception.ODatabaseException: Cannot create database 'OSystem'
            DB name="OSystem"
            at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.create(ODatabaseDocumentTx.java:450)
            at com.orientechnologies.orient.server.OSystemDatabase.init(OSystemDatabase.java:160)
            at com.orientechnologies.orient.server.OSystemDatabase.<init>(OSystemDatabase.java:44)
            at com.orientechnologies.orient.server.OServer.initSystemDatabase(OServer.java:1222)
            at com.orientechnologies.orient.server.OServer.activate(OServer.java:342)
            at com.orientechnologies.orient.server.OServerMain.main(OServerMain.java:41)
    Caused by: com.orientechnologies.orient.core.exception.OStorageException: Cannot create folders in storage with path /usr/local/var/db/orientdb/OSystem
            DB name="OSystem"
            at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.create(OLocalPaginatedStorage.java:118)
            at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.create(ODatabaseDocumentTx.java:402)
            ... 5 more
    ```

That's about as far as I got. Maybe there are more problems and more stuff needs to be adjusted.

**Comment by austinsmorris** [comment]
Great, thanks for the input!  I'll keep working on it.

File: Formula/orientdb.rb [hunk 2]
```diff
@@ -27,9 +26,15 @@ def install
     chmod 0755, Dir["bin/*"]
     libexec.install Dir["*"]
 
-    mkpath "#{libexec}/log"
-    touch "#{libexec}/log/orientdb.err"
-    touch "#{libexec}/log/orientdb.log"
+    mkpath "#{var}/db/orientdb"
```
Comment (ilovezfs) [inline Formula/orientdb.rb]: should probably make this (and the one below)
```
(var/"db/orientdb").mkpath
```

File: Formula/orientdb.rb [hunk 3]
```diff
@@ -27,9 +26,15 @@ def install
...
+    mkpath "#{var}/log/orientdb"
+    touch "#{var}/log/orientdb/orientdb.err"
+    touch "#{var}/log/orientdb/orientdb.log"
+    inreplace "#{libexec}/config/orientdb-server-config.xml", "</properties>", "  <entry name=\"server.database.path\" value=\"#{var}/db/orientdb\" />\n    </properties>"
+    inreplace "#{libexec}/config/orientdb-server-log.properties", "../log", "#{var}/log/orientdb"
+    inreplace "#{libexec}/bin/orientdb.sh", "../log", "#{var}/log/orientdb"
+    inreplace "#{libexec}/bin/server.sh", "ORIENTDB_PID=$ORIENTDB_HOME/bin", "ORIENTDB_PID=#{var}/run"
```
Comment (ilovezfs) [inline Formula/orientdb.rb]: probably should use an orientdb subdirectory in var/run.

**Comment by austinsmorris** [comment]
@ilovezfs, thanks for the suggestions, I've made those changes.  Do you know how I can stop orientdb from writing to `/usr/local/var/db/orientdb` during test?  This causes the tests to fail.

**Comment by MikeMcQuaid** [comment]
@austinsmorris That'd be `orientdb` specific but presumbly an environment variable or configuration file.

**Comment by UniqMartin** [comment]
> Do you know how I can stop orientdb from writing to `/usr/local/var/db/orientdb` during test? This causes the tests to fail.

Ideally you would want to make sure that it also has its own PID file and log files as its undesirable that the test modifies these if the test every happens to be run by a user (contrary to our CI that always starts from a clean state). For the parts where you modify the shell scripts, something like that can be achieved by allowing those values to be overridden by environment variables and falling back to the defaults if unset, e.g.:

```ruby
inreplace "#{libexec}/bin/server.sh", "ORIENTDB_PID=$ORIENTDB_HOME/bin", "ORIENTDB_PID=\"${ORIENTDB_PID:-#{var}/run/orientdb}\""
```

and then setting the environment variable in the `test do` block like this:

```ruby
ENV["ORIENTDB_PID"] = testpath/"pid"
```

For the other two files that you need to manipulate for the regular installation, you'd need to check how to achieve this. Maybe you can copy the configuration files to the test path and then tell the server to use those files instead of the ones in the usual location.

**Comment by SlamJam** [comment]
@austinsmorris for successful passing of the tests use the following patch:

```diff
diff --git a/Formula/orientdb.rb b/Formula/orientdb.rb
index fb55488..48764c0 100644
--- a/Formula/orientdb.rb
+++ b/Formula/orientdb.rb
@@ -28,6 +28,7 @@ class Orientdb < Formula
 
     (var/"db/orientdb").mkpath
     (var/"log/orientdb").mkpath
+    (var/"run/orientdb").mkpath
     touch "#{var}/log/orientdb/orientdb.err"
     touch "#{var}/log/orientdb/orientdb.log"
     inreplace "#{libexec}/config/orientdb-server-config.xml", "</properties>", "  <entry name=\"server.database.path\" value=\"#{var}/db/orientdb\" />\n    </properties>"
@@ -46,8 +47,13 @@ class Orientdb < Formula
   end
 
   test do
+     ENV["CONFIG_FILE"] = "#{testpath}/orientdb-server-config.xml"
+
+     cp "#{libexec}/config/orientdb-server-config.xml", testpath
+     inreplace "#{testpath}/orientdb-server-config.xml", "</properties>", "  <entry name=\"server.database.path\" value=\"#{testpath}\" />\n    </properties>"
+
     system "#{bin}/orientdb", "start"
-    sleep 2
+    sleep 4
 
     begin
       assert_match "OrientDB Server v.2.2.2", shell_output("curl -I localhost:2480")

```

**Comment by austinsmorris** [comment]
Thanks @SlamJam, looks like it's good to go!

File: Formula/orientdb.rb [hunk 4]
```diff
@@ -27,9 +26,16 @@ def install
...
 
-    mkpath "#{libexec}/log"
-    touch "#{libexec}/log/orientdb.err"
-    touch "#{libexec}/log/orientdb.log"
+    (var/"db/orientdb").mkpath
+    (var/"log/orientdb").mkpath
+    (var/"run/orientdb").mkpath
+    touch "#{var}/log/orientdb/orientdb.err"
+    touch "#{var}/log/orientdb/orientdb.log"
+    inreplace "#{libexec}/config/orientdb-server-config.xml", "</properties>", "  <entry name=\"server.database.path\" value=\"#{var}/db/orientdb\" />\n    </properties>"
```
Comment (MikeMcQuaid) [inline Formula/orientdb.rb]: Use `EOS.undent` instead of `\n`s. Split this into multiple lines so it wraps at ~80 characters.

**Comment by MikeMcQuaid** [comment]
Last tiny nit and we should be good to go here 👍

**Comment by hayesdavis** [comment]
I'd love to see this get merged. Anything I could to do expedite the process?

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

**Comment by MikeMcQuaid** [comment]
Merged and then cleaned up in b89772f so people could get using this sooner rather than later.

**Pull request closed without merging**